dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/JaxRsEndpoints.java

```diff
@@ -331,6 +341,75 @@ public Response getJax(@Context UriInfo uriInfo) {
         return Response.ok(Span.current().getSpanContext().getTraceId()).build();
     }
 
+    @GET
+    @Path("/contextPropagatorTwo")
+    public Response contextPropagatorTwo(@Context HttpHeaders headers) throws Exception {
+        String traceState = Span.current().getSpanContext().getTraceState().toString();
+        if(traceState.contains("entries=[test, tracestate]")){
+            return Response.ok("Tracestate found").build();
+        }
+        return Response.ok("Tracestate not found").build();
```

Comment:
Can we inspect the tracestate more directly rather than looking at the toString please?